collector/mongodb_collector.go

```diff
@@ -37,8 +37,8 @@ func NewMongodbCollector(opts MongodbCollectorOpts) *MongodbCollector {
 func (exporter *MongodbCollector) Describe(ch chan<- *prometheus.Desc) {
 	glog.Info("Describing groups")
 	session := shared.MongoSession(exporter.Opts.URI)
-	defer session.Close()
 	if session != nil {
+		defer session.Close()
```

Comment:
Since this is a simple function with only one exit point, using just `session.Close()` at the end instead of a `defer` will have less overhead. Usually `defer` adds some overhead and it is bigger in older Go versions.